File: tests/pages/firefox/whatsnew/whatsnew_73.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage


class FirefoxWhatsNew73Page(BasePage):

    URL_TEMPLATE = '/{locale}/firefox/73.0/whatsnew/all/{params}'

    _set_default_button_locator = (By.ID, 'set-as-default-button')

    @property
    def is_default_browser_button_displayed(self):
        return self.is_element_displayed(*self._set_default_button_locator)

File: media/js/firefox/whatsnew/whatsnew-73.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function() {
    'use strict';

    Mozilla.UITour.getConfiguration('appinfo', function(details) {
        if (details.defaultBrowser) {
            document.querySelector('body').classList.add('is-firefox-default');
        }
    });

    // modal
    var content = document.querySelector('.mzp-u-modal-content');
    var trigger = document.querySelector('.js-modal-link');

    trigger.addEventListener('click', function(e) {
        e.preventDefault();
        Mzp.Modal.createModal(this, content, {
            closeText: window.Mozilla.Utils.trans('global-close'),
        });

        window.dataLayer.push({
            'event': 'in-page-interaction',
            'eAction': 'link click',
            'eLabel': 'Get Firefox for mobile'
        });
    }, false);

    // initialize send to device widget
    var form = new Mozilla.SendToDevice();
    form.init();

})(window.Mozilla);
